design/authenticators/authenticator_management_improvement.md

```diff
@@ -0,0 +1,247 @@
+  - [Drawbacks](#drawbacks)
+  - [Implementation Options](#implementation-options)
+    - [Alternative 1 - Server Side](#alternative-1---server-side)
+      - [Pros](#pros)
+      - [Cons](#cons)
+    - [Alternative 2 - CLI Side](#alternative-2---cli-side)
+      - [Pros](#pros-1)
+      - [Cons](#cons-1)
+  - [Version Update](#version-update)
+  - [Delivery Plan](#delivery-plan)
+  
+## Introduction
+
+Conjur authenticators are defined as sub-policies, under the `conjur` policy.  
+Currently to define an authenticator, the user needs to complete the following steps:
+
+- Read the documentation
+- Understand how the authenticator policy is structured
+- Create a policy yaml file that defines the authenticator
+- Load the policy using Conjur CLI
+- Enable the authenticator through an environment variable or REST API call
+
+To delete an authenticator, the user needs to complete these steps.  
+Either:
+
+- Create a policy with a `- !delete` statement, on the authenticator policy
+- Load the policy using Conjur CLI
+
+Or:
+
+- Update the `conjur` policy yaml file and take out the authenticator
+- Reload the `conjur` policy using the `--replace` option  
+
+The current user experience requires a certain level of expertise. The user needs to understand the policy structure of each authenticator and the process is comprised of multiple steps. This proposal offers an easier way for managing authenticators.
+
+## Proposed Solution
+
+The proposed solution is to add a few simple Conjur CLI commands (and possibly APIs, see two alternatives below), that would hide the complexity and provide a single step to perform each action:
+
+### Add Authenticator
+
+#### Functionality
+
+Defines a new authenticator in Conjur. Behind the scenes, this command will creat the authenticator policy and fill it with all its required data. If the data is not required to be given by the user, like the `authn-k8s` CA cert and key, the information will be generated automatically.
+
+#### Synopsis
+
+conjur config authenticator add *authenticator-type/service-id* [*parameters*]
+
+#### Example
+
+```shell script
+$ conjur config authenticator add authn-oidc/my-idp --provider-uri my-uri --id-token-user-property prop
+```
+
+### Delete Authenticator
+
+#### Functionality
+
+Deletes an existing authenticator in Conjur. Deletes the authenticator policy.
+
+#### Synopsis
+
+conjur config authenticator delete *authenticator-type/service-id*
+
+#### Example
+
+```shell script
+$ conjur config authenticator delete authn-oidc/my-idp
+```
+
+### Update Authenticator
+
+#### Functionality
+
+Updates an existing authenticator in Conjur. Updates the authenticator variables.
+
+#### Synopsis
+
+conjur config authenticator update *authenticator-type/service-id* [*parameters*]
+
+#### Example
+
+```shell script
+$ conjur config authenticator update authn-oidc/my-idp --provider-uri new-uri
+```
+
+### Enable Authenticator
+
+#### Functionality
+
+Enables an authenticator in Conjur.
```

Comment:
As we have 2 methods of enablement it's best to specify where this will enable the authenticator. I guess that the DB is the most suitable via the CLI

Reply:
Yes, that was my thought. I'll specify it, it's worth mentioning to create a shared understanding.